### ISSUE TITLE
no-duplicate-selectors false positive

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#968]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`no-duplicate-selectors` is reporting false positives when run from within Atom's linter-stylelint. Originally reported under https://github.com/AtomLinter/linter-stylelint/issues/138 by @revyh.

Everything appears like it is being fed in correctly:
![image](https://cloud.githubusercontent.com/assets/427137/14027675/18679c4a-f1b6-11e5-9f71-c3712a2b5e9b.png)

> Which rule, if any, is this issue related to?

`no-duplicate-selectors`, possibly others.

> What CSS is needed to reproduce this issue?

```css
.blah {
  color: red;
}
```

The exact files used: [linter-stylelint_GH138.zip](https://github.com/stylelint/stylelint/files/188875/linter-stylelint_GH138.zip)

> What stylelint configuration is needed to reproduce this issue?

```yaml
---
  rules:
    no-duplicate-selectors: [true, {severity: error}]
```

> Which version of stylelint are you using?

`5.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Via the Node API within `linter-stylelint`.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No false positive.

> What actually happened (e.g. what warnings or errors you are getting)?

![image](https://cloud.githubusercontent.com/assets/427137/14034330/8e0d345e-f1df-11e5-984b-34a8e0996c76.png)

**Comment by davidtheclark** [comment]
I can't reproduce. In 3e4c3d2761cbfac0830b32e726ec6bbd26f706e5 I added `no-duplicate-selectors` to the integration test and did not get any fast positives. I will guess I will try to debug in Atom.

**Comment by davidtheclark** [comment]
That was a weird one. Must happen in Atom linter because a new process is not started each time stylelint runs, so a certain Map() declared outside the scope of the rule's function retained the values it had from the first run when you did the second run --- so it looked like a duplicate selector. Will release this fix soon.

**Comment by davidtheclark** [comment]
@Arcanemagus should be fixed in 5.2.0

**Comment by revyh** [comment]
@Arcanemagus @davidtheclark Thank you for the quick response!

**Comment by Arcanemagus** [comment]
Wow, that really was a bizarre one :open_mouth:. Publishing a release incorporating this soon!
